core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -470,34 +472,50 @@ private fun setFlashModeInternal(
     Log.d(TAG, "Set flash mode to: ${imageCapture.flashMode}")
 }
 
-private suspend fun startVideoRecordingInternal(
-    initialMuted: Boolean,
+private fun getPendingRecording(
+    context: Context,
     videoCaptureUseCase: VideoCapture<Recorder>,
     maxDurationMillis: Long,
     captureTypeSuffix: String,
-    context: Context,
     videoCaptureUri: Uri?,
     shouldUseUri: Boolean,
```

Comment:
We should really merge these into a single sealed interface to clean up the conditional code below. If we put them into something like:
```
sealed interface VideoOutput {
    data class MediaStore(val captureTypeSuffix: String) : VideoOutput
    data class File(val file: File) : VideoOutput
    data class ContentUri(val uri: Uri) : VideoOutput
}
```

Then we could get rid of the messy nullable Uri / boolean combination to specify whether we should use the Uri. It would also allow us to validate the URI well ahead of trying to start the recording.

Reply:
Let's leave this for another PR that specifically improves uri validation as we talked about